ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -1636,9 +1644,33 @@ - (BOOL)isAnswerValid:(id)answer {
 }
 
 - (BOOL)isAnswerValidWithString:(NSString *)text {
+    return ([self isTextAnswerValidWithString:text] && [self isEmailAddressValidWithString:text]);
+}
+
+- (BOOL)isTextAnswerValidWithString:(NSString *)text {
     return (_maximumLength == 0 || [text length] <= _maximumLength);
 }
 
+- (BOOL)isEmailAddressValidWithString:(NSString *)text {
+    if (self.isEmailAddress){
+        NSString *emailValidationRegex = @"[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,6}";
+        NSPredicate *emailValidationTest = [NSPredicate predicateWithFormat:@"SELF MATCHES %@",emailValidationRegex];
+        return [emailValidationTest evaluateWithObject:text];
+    } else {
+        return YES;
+    }
+}
+
+- (NSString *)localizedInvalidValueStringWithAnswerString:(NSString *)text {
+    NSString *string = nil;
+    if (_maximumLength != 0 && [text length] > _maximumLength) {
+        string = [NSString stringWithFormat:ORKLocalizedString(@"TEXT_ANSWER_EXCEEDING_MAX_LENGTH_ALERT_MESSAGE", nil), [@(_maximumLength) stringValue]];
+    } else if (self.emailAddress) {
+        string = [NSString stringWithFormat:ORKLocalizedString(@"INVALID_EMAIL_ALERT_MESSAGE", nil), text];
+    }
```

Comment:
Shall we reuse `isEmailAddressValidWithString:` and  `isTextLengthValidWithString:` in this method?